odc/geo/_crs.py

```diff
@@ -16,11 +16,28 @@
 SomeCRS = Union[str, "CRS", _CRS, Dict[str, Any]]
 MaybeCRS = Optional[SomeCRS]
 
+_crs_cache: Dict[str, Tuple[_CRS, str, Optional[int]]] = {}
 
-@cachetools.cached({})
-def _make_crs(crs_str: str) -> Tuple[_CRS, Optional[int]]:
-    crs = _CRS.from_user_input(crs_str)
-    return (crs, crs.to_epsg())
+
+def _make_crs_key(crs_spec: Union[str, _CRS]) -> str:
+    if isinstance(crs_spec, str):
+        normed_epsg = crs_spec.upper()
+        if normed_epsg.startswith("EPSG:"):
+            return normed_epsg
+        return crs_spec
+    return crs_spec.to_wkt()
+
+
+@cachetools.cached(_crs_cache, key=_make_crs_key)
```

Comment:
Would it make sense to add a `lock` in case of multi-threaded access? Globals w/o a lock feel weird to me but mabye that's just me.

Reply:
I'm not sure how cachetools deals with concurrency actually. Dictionaries are "thread-safe" in Python, and so are imports, so wrapping cache around a function should be thread-safe as it happens at import time, and then I assume cachetools does reasonable thing afterwards, which at worst will create 2 copies of the same CRS, but will only cache one.

Reply:
There is an option to supply `lock` to `@cached` decorator, also should probably replace plain `Dict` with some cache class instead.
more details here: https://cachetools.readthedocs.io/en/stable/#cachetools.cached

given that this is not part of the API, I think we can delay this change until later. Also need to check `pyproj` constraints for multi-threaded access.